index.js

```diff
@@ -98,6 +99,33 @@ var SpellChecker = {
             throw new Error("The dictionary file could not be read: " + file_path + ". Error: " + err);
         }
     },
+
+    /**
+     * Unzip a zip file.
+     *
+     * Each entry in the zip file will be extracted atomically. From the perspective of another
+     * process, the unzipped file will either not exist or will be fully unzipped.
+     *
+     * @param {String} zipPath The path of the zip file.
+     * @param {String} destinationDir The directory to unzip into.
+     * @throws An error if the file couldn't be unzipped.
+     */
+    _unzipSync: function(zipPath, destinationDir) {
+        // Unzip into a tmp directory.
+        var tmpDir = tmp.dirSync();
+        var zip = new Zip(zipPath);
+        zip.extractAllTo(tmpDir.name);
+
+        // Move the unzipped files out of the tmp directory and into the destination directory.
+        zip.getEntries().forEach(({ entryName }) => {
+            var from = path.join(tmpDir.name, entryName);
+            var to = path.join(destinationDir, entryName);
+            fs.renameSync(from, to);
```

Comment:
This is the important part. The rename operation is atomic, so when another process [checks for the existence of the unzipped file](https://github.com/jfmdev/simple-spellchecker/pull/32/files#diff-168726dbe96b3ce427e7fedce31bb0bcR146), it'll either see the full file or nothing at all.